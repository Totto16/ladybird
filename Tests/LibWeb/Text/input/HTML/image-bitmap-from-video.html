<video id="vid" width="640" height="480" src="file:///home/totto/Downloads/p4/Lucifer_NCIS[S06E01].mp4"></video>
<script src="../include.js"></script>
<script>
    asyncTest(async done => {
        try {
            const v = document.createElement("video");
            const result = await createImageBitmap(v, 0, 0, 20, 20);
            println(`First result: ${result}`);
        } catch (err) {
            println(`First error: ${err}`);
        }

        try {
            const v = document.getElementById("vid");
            await new Promise(resolve => {
                if (v.readyState >= HTMLMediaElement.HAVE_CURRENT_DATA) {
                    resolve();
                }
                v.addEventListener("loadeddata", () => {
                    if (v.readyState >= HTMLMediaElement.HAVE_CURRENT_DATA) {
                        resolve();
                    }
                });
            });
            const result = await createImageBitmap(v, 0, 0, 20, 20);
            println(`Second result: ${result}`);
        } catch (err) {
            println(`Second error: ${err}`);
        }

        done();
    });
</script>
