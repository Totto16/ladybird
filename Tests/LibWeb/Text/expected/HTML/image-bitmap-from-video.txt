First error: InvalidStateError: image argument is not usable
Second result: [object ImageBitmap]
